packages/react/src/primitives/shared/utils.ts

```diff
@@ -159,3 +159,15 @@ export const mergeVariantsAndOverrides = (
     ...merged,
   };
 };
+
+export type Modifiers = string | number | null;
+export const classNameModifier = (
```

Comment:
Can we have comments (or is this the best place to doc this...?) that explain how/why this is used or needed? Just looking through the uses of it for each component and they generally follow the same pattern of:

```
className={classNames(
  ComponentClassNames.ComponentName,
  classNameModifier(ComponentClassNames.ComponentName, size),
  className
)}
```

So I'm wondering if we should have a small blurb to say if these need to be in a specific order or if there is any additional nuance there. If I were to create a brand new component that is something I would be curious about.

Reply:
good call, I've added comments to clarify what this function is and how it is used, also to simplify things I've separated the two use cases into two functions.